src/Symfony/Component/Webhook/Server/JsonBodyConfigurator.php

```diff
@@ -20,14 +20,22 @@
  */
 final class JsonBodyConfigurator implements RequestConfiguratorInterface
 {
-    public function __construct(
-        private readonly SerializerInterface $serializer,
-    ) {
+    private PayloadEncoderInterface $payloadEncoder;
+
+    public function __construct(SerializerInterface|PayloadEncoderInterface|null $payloadEncoder = null)
+    {
+        if (null === $payloadEncoder) {
+            $this->payloadEncoder = new JsonPayloadEncoder();
+        } elseif ($payloadEncoder instanceof SerializerInterface) {
+            $this->payloadEncoder = new SerializerPayloadEncoder($payloadEncoder);
```

Comment:
We could consider deprecating passing a serializer instance, now that we have a dedicated contract for this dependency.

Reply:
I thought about that as well, but as I don't see any downside in using the serializer directly I didn't want to impose the need to change anything on users of the component.